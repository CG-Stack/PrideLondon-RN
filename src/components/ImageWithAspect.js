// @flow
import React, { Component } from "react";
import { Image, View } from "react-native";

type Props = {
  ratio: Number,
  source: Image.Props.source
};

class ImageWithAspect extends Component<Props> {
  constructor() {
    super();
    this.state = {
      height: 0
    };
  }
  handleLayout = event => {
    this.setState({
      height: event.nativeEvent.layout.width / this.props.ratio
    });
  };

  render() {
    return (
      <View onLayout={this.handleLayout}>
        <Image
          source={this.props.source}
          style={{
            height: this.state.height
          }}
          resizeMode="cover"
        />
      </View>
    );
  }
}

export default ImageWithAspect;
