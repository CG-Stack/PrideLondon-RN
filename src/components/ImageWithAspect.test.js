// @flow
import React from "react";
import { shallow } from "enzyme";
import ImageWithAspect from "./ImageWithAspect";

it("renders correctly", () => {
  const output = shallow(
    <ImageWithAspect
      ratio={2 / 1}
      source={{ uri: "/super-awesome-image.jpg" }}
    />
  );
  expect(output).toMatchSnapshot();
});

it("recalcualtes width on layout event", () => {
  const output = shallow(
    <ImageWithAspect
      ratio={2 / 1}
      source={{ uri: "/super-awesome-image.jpg" }}
    />
  );
  output.simulate("layout", { nativeEvent: { layout: { width: 100 } } });
  expect(output.state("height")).toEqual(50);
  expect(output).toMatchSnapshot();
});
