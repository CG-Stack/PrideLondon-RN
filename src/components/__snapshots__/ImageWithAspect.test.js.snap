// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recalcualtes width on layout event 1`] = `
<View
  onLayout={[Function]}
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "/super-awesome-image.jpg",
      }
    }
    style={
      Object {
        "height": 50,
      }
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "/super-awesome-image.jpg",
      }
    }
    style={
      Object {
        "height": 0,
      }
    }
  />
</View>
`;
